internal/testhelpers/e2e/spawn.go

```diff
@@ -1,47 +1,174 @@
 package e2e
 
-type SpawnOptions func(*Options) error
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
 
-func WithArgs(args ...string) SpawnOptions {
-	return func(opts *Options) error {
-		opts.Options.Args = args
-		return nil
+	"github.com/ActiveState/termtest"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/osutils"
+)
+
+type SpawnedCmd struct {
+	*termtest.TermTest
+	opts SpawnOpts
+}
+
+func (s *SpawnedCmd) WorkDirectory() string {
+	return s.TermTest.Cmd().Dir
+}
+
+func (s *SpawnedCmd) Wait() error {
+	return s.TermTest.Wait(30 * time.Second)
+}
+
+func (s *SpawnedCmd) Executable() string {
+	return s.TermTest.Cmd().Path
+}
+
+// StrippedSnapshot returns the snapshot with trimmed whitespace and stripped line endings
+// Mainly intended for JSON parsing
+func (s *SpawnedCmd) StrippedSnapshot() string {
+	return strings.Trim(strings.ReplaceAll(s.TermTest.Snapshot(), "\n", ""), "\x00\x20\x0a\x0d")
+}
+
+func (s *SpawnedCmd) ExpectRe(v string, opts ...termtest.SetExpectOpt) error {
```

Comment:
The reason it's like this is so that we can fail the regex through the failure chain of termtest, which extends to the testing failure chain. 

If we just rely on MustCompile then it will fail with a panic rather than a more elaborate and informative test failure.

Reply:
Will add a comment to explain this rationale.